util/src/test/java/io/kubernetes/client/util/generic/GenericKubernetesApiForCoreApiTest.java

```diff
@@ -169,6 +169,36 @@ public void patchNamespacedPodReturningObject() {
     verify(1, patchRequestedFor(urlPathEqualTo("/api/v1/namespaces/default/pods/foo1")));
   }
 
+  @Test
+  public void patchNamespacedPodInRancher() {
+    V1Patch v1Patch = new V1Patch("{}");
+    V1Pod foo1 =
+        new V1Pod().kind("Pod").metadata(new V1ObjectMeta().namespace("default").name("foo1"));
+    // add url prefix
+    String prefix = "/k8s/clusters/c-7q988";
+    stubFor(
+        patch(urlEqualTo(prefix + "/api/v1/namespaces/default/pods/foo1"))
+            .withHeader("Content-Type", containing(V1Patch.PATCH_FORMAT_STRATEGIC_MERGE_PATCH))
+            .willReturn(aResponse().withStatus(200).withBody(new Gson().toJson(foo1))));
+
+    GenericKubernetesApi<V1Pod, V1PodList> rancherPodClient =
+        new GenericKubernetesApi<>(
+            V1Pod.class,
+            V1PodList.class,
+            "",
+            "v1",
+            "pods",
+            new ClientBuilder().setBasePath("http://localhost:" + 8181 + prefix).build());
```

Comment:
on the other hand, i think the usage of `setBasePath` here is error-proning b/c the setter is doing an in-place modification upon the `ApiClient` instance. there will be thread-safety issue if multiple generic api modifying the basepath concurrently. ideally, what i expected is a function to provide an immutable shallow copy of `ApiClient` before we tweak the prefix, such as:

```
ApiClient shallowCopied = apiClient.newBuilder().setBasePath("<path>" + prefix).build()
```

we can do this in a follow-up anyway

Reply:
> i think the usage of `setBasePath` here is error-proning b/c the setter is doing an in-place modification upon the `ApiClient` instance. there will be thread-safety issue if multiple generic api modifying the basepath concurrently

Sorry I didn't get your point, there is no method for `GenericKuberentesApi` to modify the basepath, each `GenericKuberentesApi` has its own `ApiClient`. Please correct me if I'm wrong.

Do you mean multiple `GenericKuberentesApi` sharing one `ApiClient`?

Reply:
oh oversight, i took `ClientBuilder` as `ApiClient`.